shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala

```diff
@@ -395,6 +395,12 @@ object GlutenConfig {
       k => {
         if (conf.contains(k)) {
           nativeConfMap.put(k, conf(k))
+        } else {
+          // Session's local time zone must be set. If not explicitly set by user,
+          // its default value is used, consistent with spark.
+          if (k.equals(SQLConf.SESSION_LOCAL_TIMEZONE.key)) {
+            nativeConfMap.put(k, SQLConf.SESSION_LOCAL_TIMEZONE.defaultValueString)
```

Comment:
Maybe we need to verify if there are other cases with similar problems?

Reply:
After some checks, it looks there is no other similar issue. Only timezone config needs special handling as of now. Thanks!